microdata_validator/schema/metadata.py

```diff
@@ -0,0 +1,174 @@
+import datetime
+from enum import Enum
+from typing import Optional, List, Union
+
+from pydantic import BaseModel, conlist, root_validator, Extra
+
+
+class TemporalityType(str, Enum):
+    FIXED = "FIXED"
+    STATUS = "STATUS"
+    ACCUMULATED = "ACCUMULATED"
+    EVENT = "EVENT"
+
+
+class DataType(str, Enum):
+    STRING = "STRING"
+    LONG = "LONG"
+    DATE = "DATE"
+    DOUBLE = "DOUBLE"
+
+
+class SensitivityLevel(str, Enum):
+    PERSON_GENERAL = "PERSON_GENERAL"
+    PERSON_SPECIAL = "PERSON_SPECIAL"
+    PUBLIC = "PUBLIC"
+    NONPUBLIC = "NONPUBLIC"
+
+
+class LanguageCode(str, Enum):
+    no = "no"
+    nb = "nb"
+    nn = "nn"
+    en = "en"
+
+
+class UnitType(str, Enum):
+    JOBB = "JOBB"
+    KJORETOY = "KJORETOY"
+    FAMILIE = "FAMILIE"
+    FORETAK = "FORETAK"
+    HUSHOLDNING = "HUSHOLDNING"
+    KOMMUNE = "KOMMUNE"
+    KURS = "KURS"
+    PERSON = "PERSON"
+    VIRKSOMHET = "VIRKSOMHET"
+
+
+class MultiLingualString(BaseModel):
+    languageCode: LanguageCode
+    value: str
+
+
+class DataRevision(BaseModel, extra=Extra.forbid):
+    description: conlist(MultiLingualString, min_items=1)
+    temporalEndOfSeries: bool
+
+
+class IdentifierVariable(BaseModel, extra=Extra.forbid):
+    unitType: UnitType
+
+
+class CodeListItem(BaseModel, extra=Extra.forbid):
+    code: str
+    categoryTitle: conlist(MultiLingualString, min_items=1)
+    validFrom: str
+    validUntil: Optional[Union[str, None]]
+
+    @root_validator(skip_on_failure=True)
```

Comment:
Seems that `skip_on_failure` is not needed here as we don't have field validators in this class. [Ref](https://docs.pydantic.dev/usage/validators/#root-validators)

Reply:
True 👍🏻 I have mistakenly understood "field validators" as the typing itself, but you are right. Will fix 🏃🏻

Reply:
Checked now and the "field validators" actually does include the typing itself. So these should be skipped if the typing is wrong.

Example:
* You add an "extraField" inside a codelist, which in turn makes it fail its type validation as a codelist Item
* This gets reported to user
* But if the root validator runs after it will also report that you are missing a codeList (since it failed the type validation)

So it seems skip_on_failure means that it will get skipped if the type validators are not in order.

Reply:
All right 👍